skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,367 @@
+        dist_ngboost = Normal
+        if self.dist in ngboost_dists:
+            dist_ngboost = ngboost_dists[self.dist]
+
+        return dist_ngboost
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from ngboost import NGBRegressor
+        from sklearn.tree import DecisionTreeRegressor
+
+        if self.estimator is None:
+            self.estimator = DecisionTreeRegressor(
+                criterion="friedman_mse",
+                min_samples_split=2,
+                min_samples_leaf=1,
+                min_weight_fraction_leaf=0.0,
+                max_depth=3,
+                splitter="best",
+                random_state=None,
+            )
+
+        dist_ngboost = self.dist_to_ngboost_instance()
+
+        self.ngb = NGBRegressor(
+            Dist=dist_ngboost,
+            Score=self.score,
+            Base=self.estimator,
+            natural_gradient=True,
+            n_estimators=self.n_estimators,
+            learning_rate=self.learning_rate,
+            minibatch_frac=self.minibatch_frac,
+            col_sample=self.col_sample,
+            verbose=self.verbose,
+            verbose_eval=self.verbose_eval,
+            tol=self.tol,
+            random_state=self.random_state,
+            validation_fraction=self.validation_fraction,
+            early_stopping_rounds=self.early_stopping_rounds,
+        )
+        self.ngb.fit(X, y)
+        return self
+
+    def _predict(self, X):
+        return self.ngb.predict(X)
+
+    def _pred_dist(self, X):
+        return self.ngb.pred_dist(X)
+
+    def _ngb_dist_to_skpro(self, pred_mean, pred_std, index, columns):
+        """
+        Convert NGBoost distribution object to skpro BaseDistribution object.
+
+        Parameters
+        ----------
+        pred_mean, pred_std and index and columns.
+
+        Returns
+        -------
+        skpro_dist (skpro.distributions.BaseDistribution):
+        Converted skpro distribution object.
+        """
+        from skpro.distributions.laplace import Laplace
+        from skpro.distributions.lognormal import LogNormal
+        from skpro.distributions.normal import Normal
+        from skpro.distributions.poisson import Poisson
+        from skpro.distributions.t import TDistribution
+
+        ngboost_dists = {
+            "Normal": Normal,
+            "Laplace": Laplace,
+            "TDistribution": TDistribution,
+            "Poisson": Poisson,
+            "LogNormal": LogNormal,
+        }
+
+        skpro_dist = None
+
+        mu_sigma = ["Normal", "TDistribution", "LogNormal"]
```

Comment:
you do this translation to and from pred_mean etc twice, once from and once to.

It would reduce unnecessary complexity of the code if, instead, you would have the parameter names per distribution in a dict, pass the params to this method, and use `**kwargs` to pass it to `ngboost_dists[self.dist]`.

Reply:
I have implemented it in this way now please let me know if it needs to be improved further.